tests/rptest/tests/leaders_info_api_test.py

```diff
@@ -0,0 +1,38 @@
+# Copyright 2022 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+from rptest.services.cluster import cluster
+from rptest.clients.types import TopicSpec
+from rptest.services.admin import Admin
+from rptest.tests.redpanda_test import RedpandaTest
+from rptest.services.rpk_producer import RpkProducer
+
+
+class LeadersInfoApiTest(RedpandaTest):
+    topics = (TopicSpec(partition_count=3, replication_factor=3),
+              TopicSpec(partition_count=3, replication_factor=3),
+              TopicSpec(partition_count=3, replication_factor=3))
+
+    def __init__(self, test_context):
+        super(LeadersInfoApiTest, self).__init__(test_context=test_context,
+                                                 num_brokers=3)
+
+        self.admin = Admin(self.redpanda)
+
+    @cluster(num_nodes=4)
+    def rest_leaders_info_test(self):
+        for node in self.redpanda.nodes:
```

Comment:
I would make the test using `EndToEnd` test, then wait for a while until some messages are produced, reset leader info and then use EndToEnd test verify method